### PR TITLE
Id proofs popup

### PR DESCRIPTION
**Pull request by soerenbf**
## Purpose

Present ID statement requested from dApp in the wallet and generate zero knowledge proof.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: packages/browser-wallet/src/popup/pages/IdProofRequest/DisplayStatement/utils.ts [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+    if (statement.lower === MIN_DATE) {
+        return upperInYears;
+    }
+    if (statement.upper > today) {
+        return lowerInYears;
+    }
+
+    return upperInYears && lowerInYears;
+};
```
[the remark this answers is not in the document]
Comment (soerenbf) [reply packages/browser-wallet/src/popup/pages/IdProofRequest/DisplayStatement/utils.ts:39]: Comparison of month/date parts of bounds with today to see if the difference is in years?

[the remark this answers is not in the document]
Comment (shjortConcordium) [reply packages/browser-wallet/src/popup/pages/IdProofRequest/DisplayStatement/utils.ts:39]: isUpperYearOffset

File: packages/browser-wallet/src/popup/pages/IdProofRequest/DisplayStatement/utils.ts [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+                if (statement.upper > today) {
+                    return t('dobMin', { dobString: minDobString });
+                }
+
+                return t('dobBetween', { minDobString, maxDobString });
+            }
+            case 'idDocIssuedAt':
+                return t('idValidity', { dateString: formatDateString(statement.upper) });
+            case 'idDocExpiresAt':
+                return t('idValidity', { dateString: formatDateString(statement.lower) });
```
[the remark this answers is not in the document]
Comment (soerenbf) [reply packages/browser-wallet/src/popup/pages/IdProofRequest/DisplayStatement/utils.ts]: Discussed with @jens-concordium , and we made something more generic.

File: packages/browser-wallet/src/popup/pages/IdProofRequest/DisplayStatement/DisplayStatement.tsx [hunk 3]
```diff
@@ -0,0 +1,266 @@
+/* eslint-disable react/prop-types */
```
Comment (shjortConcordium) [inline packages/browser-wallet/src/popup/pages/IdProofRequest/DisplayStatement/DisplayStatement.tsx:1]: Screenshot from 2022-12-07 12-46-04
:thinking:

Comment (shjortConcordium) [reply packages/browser-wallet/src/popup/pages/IdProofRequest/DisplayStatement/DisplayStatement.tsx:1]: For:
Attribute in range
dob
Lower: 19000505
Upper: 20000505

File: packages/browser-wallet-api-helpers/src/wallet-api-types.ts [hunk 4]
```diff
@@ -101,6 +103,15 @@ interface MainWalletApi {
         contractIndex: bigint,
         contractSubindex?: bigint
     ): Promise<string[]>;
+
+    /**
+     * Request that the user provides a proof for the given statement.
+     * @param accountAddress the address of the account that should prove the statement.
+     * @param statement the id statement that should be proven.
+     * @param challenge bytes chosen by the verifier. Should be HEX encoded.
+     * @returns The id proof and the credential used to prove it.
```
Comment (shjortConcordium) [inline packages/browser-wallet-api-helpers/src/wallet-api-types.ts:112]:
```suggestion
     * @returns The id proof and the id of the credential used to prove it.
```

File: packages/browser-wallet/src/popup/pages/IdProofRequest/DisplayStatement/DisplayStatement.tsx (hunk 3 above)
Comment (soerenbf) [reply packages/browser-wallet/src/popup/pages/IdProofRequest/DisplayStatement/DisplayStatement.tsx:1]: Fixed, sorry!

**Pull request merged**